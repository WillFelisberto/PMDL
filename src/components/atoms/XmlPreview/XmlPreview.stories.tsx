import type { Meta, StoryObj } from '@storybook/react';

import { XmlPreview } from './XmlPreview';

const meta: Meta<typeof XmlPreview> = {
  title: 'Atoms/XmlPreview',
  component: XmlPreview,
  tags: ['autodocs'],
  args: {
    children: 'Example',
    variant: 'default'
  }
};

export default meta;

type Story = StoryObj<typeof XmlPreview>;

export const Default: Story = {};
