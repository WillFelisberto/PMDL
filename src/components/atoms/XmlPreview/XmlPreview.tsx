'use client';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { atomDark } from 'react-syntax-highlighter/dist/cjs/styles/prism';

type XmlPreviewProps = {
  xml: string;
};

export const XmlPreview = ({ xml }: XmlPreviewProps) => {
  return (
    <div className="mt-8 space-y-4">
      <h3 className="text-lg font-semibold text-gray-700">Preview do XML PMDL</h3>

      <div className="rounded-lg overflow-hidden border border-gray-200 relative">
        <button
          onClick={() => navigator.clipboard.writeText(xml)}
          className="absolute right-2 top-2 bg-gray-800 text-white px-3 py-1 rounded-md text-sm hover:bg-gray-700 transition-colors"
        >
          Copiar
        </button>

        <SyntaxHighlighter
          language="xml"
          style={atomDark}
          data-testid="xml-preview"
          customStyle={{
            padding: '1.5rem',
            fontSize: '0.875rem',
            lineHeight: '1.5',
            margin: 0
          }}
        >
          {xml}
        </SyntaxHighlighter>
      </div>
    </div>
  );
};
