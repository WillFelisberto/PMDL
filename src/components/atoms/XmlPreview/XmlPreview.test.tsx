import { XmlPreview } from './XmlPreview';

import { render, screen } from '@/tests/test-utils';

describe('XmlPreview', () => {
  const mockXml = `<pricing-model>
  <qualifier>
    <equals>
      <value>order.total</value>
      <constant>
        <data-type>number</data-type>
        <string-value>100</string-value>
      </constant>
    </equals>
  </qualifier>
</pricing-model>`;

  it('Deve renderizar corretamente o tÃ­tulo do preview', () => {
    const { container } = render(<XmlPreview xml={mockXml} />);

    const heading = screen.getByRole('heading', { name: /preview do xml pmdl/i });
    expect(heading).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('Deve exibir o XML corretamente', async () => {
    render(<XmlPreview xml={mockXml} />);

    const xmlElement = screen.getByTestId('xml-preview');

    expect(xmlElement).toBeInTheDocument();

    expect(screen.getByText('order.total')).toBeInTheDocument();
  });
});
