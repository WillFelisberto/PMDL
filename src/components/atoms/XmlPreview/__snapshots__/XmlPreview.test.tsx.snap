// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XmlPreview Deve renderizar corretamente o t√≠tulo do preview 1`] = `
<div>
  <div
    class="mt-8 space-y-4"
  >
    <h3
      class="text-lg font-semibold text-gray-700"
    >
      Preview do XML PMDL
    </h3>
    <div
      class="rounded-lg overflow-hidden border border-gray-200 relative"
    >
      <button
        class="absolute right-2 top-2 bg-gray-800 text-white px-3 py-1 rounded-md text-sm hover:bg-gray-700 transition-colors"
      >
        Copiar
      </button>
      <pre
        data-testid="xml-preview"
        style="color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1.5rem; margin: 0px; overflow: auto; border-radius: 0.3em; background: rgb(29, 31, 33); font-size: 0.875rem;"
      >
        <code
          class="language-xml"
          style="white-space: pre; color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
        >
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            pricing-model
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
              
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            qualifier
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            equals
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                  
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            value
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            order.total
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            value
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                  
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            constant
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                    
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            data-type
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            number
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            data-type
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                    
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            string-value
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            100
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            string-value
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                  
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            constant
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
                
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            equals
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span>
              
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            qualifier
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
          <span>
            

          </span>
          <span />
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &lt;/
          </span>
          <span
            class="token"
            style="color: rgb(150, 203, 254);"
          >
            pricing-model
          </span>
          <span
            class="token"
            style="color: rgb(197, 200, 198);"
          >
            &gt;
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;
